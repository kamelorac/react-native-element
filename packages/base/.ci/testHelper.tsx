import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
