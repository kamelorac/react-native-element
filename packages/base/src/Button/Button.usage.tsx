import React from 'react';
import { Button, Icon } from '..';
import { usage, Stack, info, meta } from '@rneui/doc-gen';

info(
  'Buttons are touchable elements used to interact with the screen and to perform and operation.',
  'They may display text, icons, or both. Buttons can be styled with several props to look a specific way.'
);

meta({
  name: 'Button',
});

usage(
  'Variants',
  'There are solid button, outline button and clear button types',
  () => (
    <Stack row align="center" spacing={4}>
      <Button title="Solid" />
      <Button title="Outline" type="outline" />
      <Button title="Clear" type="clear" />
    </Stack>
  )
);

usage('Size', 'Button can be small medium or large', () => (
  <Stack row align="center" spacing={4}>
    <Button size="sm">Small</Button>
    <Button size="md">Medium</Button>
    <Button size="lg">Large</Button>
  </Stack>
));

usage('Colors', '', () => (
  <Stack row align="center" spacing={4}>
    <Button>Primary</Button>
    <Button color="secondary">Secondary</Button>
    <Button color="warning">Warning</Button>
    <Button color="error">Error</Button>
  </Stack>
));

usage('Disabled', '', () => (
  <Stack row align="center" spacing={4}>
    <Button title="Solid" disabled />
    <Button title="Outline" type="outline" disabled />
    <Button title="Clear" type="clear" disabled />
  </Stack>
));

usage('Linear Gradient', '', (LinearGradient) => (
  <Button
    ViewComponent={LinearGradient} // Don't forget this!
    linearGradientProps={{
      colors: ['#FF9800', '#F44336'],
      start: { x: 0, y: 0.5 },
      end: { x: 1, y: 0.5 },
    }}
  >
    Linear Gradient
  </Button>
));

usage(
  'Icon Button',
  'Can contain an Icon by setting the icon prop or placing an Icon component within the Button.',
  () => (
    <Button radius={'sm'} type="solid">
      Save
      <Icon name="save" color="white" />
    </Button>
  )
);

usage('Loading spinner', '', () => (
  <Button title="Solid" type="solid" loading />
));
