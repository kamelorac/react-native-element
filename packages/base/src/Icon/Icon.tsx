import React from 'react';
import {
  Platform,
  View,
  StyleSheet,
  ViewStyle,
  StyleProp,
  TextStyle,
  Pressable,
  ColorValue,
} from 'react-native';
import {
  IconButtonProps,
  IconProps as VectorIconProps,
} from 'react-native-vector-icons/Icon';
import Color from 'color';
import getIconType from '../helpers/getIconType';
import getIconStyle from '../helpers/getIconStyle';
import {
  androidRipple,
  defaultTheme,
  InlinePressableProps,
  RneFunctionComponent,
} from '../helpers';

export type IconType =
  | 'material'
  | 'material-community'
  | 'simple-line-icon'
  | 'zocial'
  | 'font-awesome'
  | 'octicon'
  | 'ionicon'
  | 'foundation'
  | 'evilicon'
  | 'entypo'
  | 'antdesign'
  | 'font-awesome-5'
  | string;

export interface IconObject {
  /** Name of icon. */
  name?: string;

  /** Color of icon. */
  color?: string;

  /** Size of icon. */
  size?: number;

  /** Type of icon */
  type?: IconType;

  /** Apply style to the icon using iconStyle. */
  iconStyle?: StyleProp<TextStyle>;
}

export type IconNode = boolean | React.ReactElement<{}> | Partial<IconProps>;

export interface IconProps extends InlinePressableProps, IconButtonProps {
  /** Type of icon set. [Supported sets here](#available-icon-sets). */
  type?: IconType;

  /** Update React Native Component.
   *  @default `Press handlers present then Pressable else View`
   */
  Component?: typeof React.Component;

  /** Reverses color scheme. */
  reverse?: boolean;

  /** Adds box shadow to button. */
  raised?: boolean;

  /** Add styling to container holding icon. */
  containerStyle?: StyleProp<ViewStyle>;

  /** Provide all props from react-native Icon component. */
  iconProps?: VectorIconProps;

  /** Specify reverse icon color. */
  reverseColor?: string;

  /** Disables onPress events. Only works when `onPress` has a handler. */
  disabled?: boolean;

  /** Style for the button when disabled. Only works when `onPress` has a handler. */
  disabledStyle?: StyleProp<ViewStyle>;

  /** Uses the solid font. */
  solid?: boolean;

  /** Uses the brands font (FontAwesome5 only). */
  brand?: boolean;
}

/** Icons are visual indicators usually used to describe action or intent.
 * They are also used for displaying information. */
export const Icon: RneFunctionComponent<IconProps> = ({
  type = 'material',
  name,
  size = 24,
  color: colorProp,
  iconStyle,
  iconProps,
  underlayColor = 'transparent',
  reverse = false,
  raised = false,
  containerStyle,
  reverseColor: reverseColorProp,
  disabled = false,
  disabledStyle,
  onPress,
  onLongPress,
  onPressIn,
  onPressOut,
  Component = onPress || onLongPress || onPressIn || onPressOut
    ? Pressable
    : View,
  solid = false,
  brand = false,
  theme = defaultTheme,
  pressableProps,
  ...rest
}) => {
  const color = colorProp || theme?.colors?.black;
  const reverseColor = reverseColorProp || theme?.colors?.white;
  const IconComponent = getIconType(type);
  const iconSpecificStyle = getIconStyle(type, { solid, brand });

  const getBackgroundColor = React.useMemo(() => {
    if (reverse) {
      return color;
    }
    return raised ? theme?.colors?.white : 'transparent';
  }, [color, raised, reverse, theme?.colors?.white]);

  const buttonStyles = React.useMemo(
    () => ({
      borderRadius: size + 4,
      height: size * 2 + 4,
      width: size * 2 + 4,
    }),
    [size]
  );

  return (
    <View
      style={StyleSheet.flatten([
        !raised && styles.container,
        (reverse || raised) && styles.button,
        (reverse || raised) && buttonStyles,
        raised && styles.raised,
        iconStyle && iconStyle.borderRadius
          ? {
              borderRadius: iconStyle.borderRadius,
            }
          : {},
        containerStyle && containerStyle,
      ])}
      testID="RNE__ICON__CONTAINER"
    >
      <Component
        testID="RNE__ICON__CONTAINER_ACTION"
        {...{
          android_ripple: androidRipple(
            Color(reverse ? color : (underlayColor as string))
              .alpha(0.3)
              .rgb()
              .string()
          ),
          onPress,
          onLongPress,
          onPressIn,
          onPressOut,
          disabled,
          accessibilityRole: 'button',
          ...pressableProps,
          ...rest,
        }}
      >
        <View
          style={StyleSheet.flatten([
            (reverse || raised) && buttonStyles,
            {
              backgroundColor: getBackgroundColor as ColorValue,
              alignItems: 'center',
              justifyContent: 'center',
            },
            disabled && styles.disabled,
            disabled && disabledStyle,
          ])}
          testID="RNE__ICON"
        >
          <IconComponent
            testID="RNE__ICON__Component"
            style={StyleSheet.flatten([
              { backgroundColor: 'transparent' },
              iconStyle && iconStyle,
            ])}
            size={size}
            name={name}
            color={reverse ? reverseColor : color}
            {...iconSpecificStyle}
            {...iconProps}
          />
        </View>
      </Component>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    overflow: 'hidden',
  },
  button: {
    margin: 7,
  },
  raised: {
    ...Platform.select({
      android: {
        elevation: 2,
      },
      default: {
        shadowColor: 'rgba(0,0,0, .4)',
        shadowOffset: { height: 1, width: 1 },
        shadowOpacity: 1,
        shadowRadius: 1,
      },
    }),
  },
  disabled: {
    backgroundColor: '#D1D5D8',
  },
});

Icon.displayName = 'Icon';
