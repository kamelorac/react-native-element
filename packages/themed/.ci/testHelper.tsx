import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';
import {
  ThemeProvider,
  FullTheme,
  createTheme,
  CreateThemeOptions,
} from '../src/config';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  themeProp: Partial<CreateThemeOptions> = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(
    <ThemeProvider theme={createTheme(themeProp)}>{children}</ThemeProvider>,
    options
  );
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
