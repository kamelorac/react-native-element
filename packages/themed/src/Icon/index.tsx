import { withTheme } from '../config';
import {
  Icon,
  IconProps,
  IconNode,
  IconObject,
  IconType,
} from '@rneui/base/dist/Icon/Icon';

export { Icon };
export type { IconProps, IconNode, IconObject, IconType };
export default withTheme<IconProps>(Icon, 'Icon');
