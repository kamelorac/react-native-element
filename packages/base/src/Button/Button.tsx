import Color from 'color';
import React, { useCallback, useEffect, useMemo } from 'react';
import {
  ActivityIndicator,
  ActivityIndicatorProps,
  Platform,
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  TouchableNativeFeedback,
  TouchableNativeFeedbackProps,
  TouchableOpacity,
  TouchableOpacityProps,
  View,
  ViewStyle,
} from 'react-native';
import {
  color,
  defaultTheme,
  renderNode,
  Theme,
  StringOmit,
  RneFunctionComponent,
  ThemeSpacing,
} from '../helpers';
import { IconNode, Icon } from '../Icon';
import { TextProps } from '../Text';

const defaultLoadingProps = (
  type: 'solid' | 'clear' | 'outline',
  theme: Theme
): ActivityIndicatorProps => ({
  color: type === 'solid' ? 'white' : theme?.colors?.primary,
  size: 'small',
});

const positionStyle = {
  top: 'column',
  bottom: 'column-reverse',
  left: 'row',
  right: 'row-reverse',
};

export interface ButtonProps
  extends TouchableOpacityProps,
    TouchableNativeFeedbackProps {
  /** Add button title. */
  title?: string | React.ReactElement<{}>;

  /** Add additional styling for title component. */
  titleStyle?: StyleProp<TextStyle>;

  /** Add additional props for Text component. */
  titleProps?: TextProps;

  /** Add additional styling for button component. */
  buttonStyle?: StyleProp<ViewStyle>;

  /** Type of button. */
  type?: 'solid' | 'clear' | 'outline';

  /** Prop to display a loading spinner. */
  loading?: boolean;

  /** Add additional styling for loading component. */
  loadingStyle?: StyleProp<ViewStyle>;

  /** Add additional props for ActivityIndicator component. */
  loadingProps?: ActivityIndicatorProps;

  /** Styling for Component container. */
  containerStyle?: StyleProp<ViewStyle>;

  /** Displays a centered icon (when no title) or to the left (with text). (can be used along with iconRight as well). Can be an object or a custom component. */
  icon?: IconNode;

  /** Styling for Icon Component container. */
  iconContainerStyle?: StyleProp<ViewStyle>;

  /** Displays Icon to the right of title. Needs to be used along with `icon` prop. */
  iconRight?: boolean;

  /** Displays a linear gradient. See [usage](#linear-gradient). */
  linearGradientProps?: object;

  /** Component for user interaction. */
  TouchableComponent?: typeof React.Component;

  /** Component for container. */
  ViewComponent?: typeof React.Component;

  /** Disables user interaction. */
  disabled?: boolean;

  /** Style of the button when disabled. */
  disabledStyle?: StyleProp<ViewStyle>;

  /** Style of the title when disabled. */
  disabledTitleStyle?: StyleProp<TextStyle>;

  /** Add raised button styling (optional). Has no effect if `type="clear"`. */
  raised?: boolean;

  /** Displays Icon to the position mentioned. Needs to be used along with `icon` prop. */
  iconPosition?: 'left' | 'right' | 'top' | 'bottom';

  /** Uppercase button title*/
  uppercase?: boolean;

  /** Radius of button
   * @type   number | sm | md | lg
   */
  radius?: number | StringOmit<keyof ThemeSpacing>;

  /** Button size */
  size?: 'sm' | 'md' | 'lg';

  /**
   * Color of Button
   * @type   string | primary | secondary | success | warning | error
   */
  color?: StringOmit<'primary' | 'secondary' | 'success' | 'error' | 'warning'>;
}

export const Button: RneFunctionComponent<ButtonProps> = ({
  TouchableComponent,
  containerStyle,
  onPress = () => {},
  buttonStyle,
  type = 'solid',
  loading = false,
  loadingStyle,
  loadingProps: passedLoadingProps,
  size = 'md',
  radius = 'xs',
  uppercase = false,
  color: buttonColor = 'primary',
  title = '',
  titleProps,
  titleStyle: passedTitleStyle,
  icon,
  iconContainerStyle,
  iconRight = false,
  disabled = false,
  disabledStyle,
  disabledTitleStyle,
  raised = false,
  linearGradientProps,
  ViewComponent = View,
  theme = defaultTheme,
  iconPosition = 'left',
  children = title,
  ...rest
}) => {
  useEffect(() => {
    if (linearGradientProps && !ViewComponent) {
      console.warn(
        "You need to pass a ViewComponent to use linearGradientProps !\nExample: ViewComponent={require('react-native-linear-gradient')}"
      );
    }
  });

  const handleOnPress = useCallback(
    (evt) => {
      if (!loading && !disabled) {
        onPress(evt);
      }
    },
    [loading, onPress, disabled]
  );

  // Refactor to Pressable
  const TouchableComponentInternal =
    TouchableComponent ||
    Platform.select({
      android: linearGradientProps ? TouchableOpacity : TouchableNativeFeedback,
      default: TouchableOpacity,
    });

  const titleStyle: StyleProp<TextStyle> = useMemo(
    () =>
      StyleSheet.flatten([
        {
          color: type === 'solid' ? 'white' : theme?.colors?.primary,
        },
        uppercase && { textTransform: 'uppercase' },
        styles.title,
        passedTitleStyle,
        disabled && {
          color: color(theme?.colors?.disabled).darken(0.3).string(),
        },
        disabled && disabledTitleStyle,
      ]),
    [
      disabled,
      disabledTitleStyle,
      passedTitleStyle,
      theme?.colors?.disabled,
      theme?.colors?.primary,
      type,
      uppercase,
    ]
  );

  const background =
    Platform.OS === 'android' && Platform.Version >= 21
      ? TouchableNativeFeedback.Ripple(
          Color(titleStyle?.color?.toString()).alpha(0.32).rgb().string(),
          false
        )
      : undefined;

  const loadingProps: ActivityIndicatorProps = useMemo(
    () => ({
      ...defaultLoadingProps(type, theme),
      ...passedLoadingProps,
    }),
    [passedLoadingProps, theme, type]
  );

  const accessibilityState = useMemo(
    () => ({
      disabled: !!disabled,
      busy: !!loading,
    }),
    [disabled, loading]
  );

  const borderRadius = useMemo(
    () =>
      Number(
        theme.spacing[radius as keyof typeof theme.spacing] ?? (radius || '0')
      ) || 0,
    [radius, theme]
  );

  return (
    <View
      style={[
        styles.container,
        { borderRadius },
        containerStyle,
        raised && !disabled && type !== 'clear' && styles.raised,
      ]}
      testID="RNE_BUTTON_WRAPPER"
    >
      <TouchableComponentInternal
        onPress={handleOnPress}
        delayPressIn={0}
        activeOpacity={0.3}
        accessibilityRole="button"
        accessibilityState={accessibilityState}
        disabled={disabled}
        background={background}
        {...rest}
      >
        <ViewComponent
          {...linearGradientProps}
          style={StyleSheet.flatten([
            styles.button,
            {
              padding: theme.spacing[size],
              paddingHorizontal: theme.spacing[size] + 2,
              borderRadius,
              // flex direction based on iconPosition
              // if iconRight is true, default to right
              flexDirection:
                positionStyle[iconRight ? 'right' : iconPosition] || 'row',
              backgroundColor:
                type === 'solid'
                  ? theme.colors[buttonColor as PropertyKey] ||
                    buttonColor ||
                    theme?.colors?.primary
                  : 'transparent',
              borderColor: theme?.colors?.primary,
              borderWidth: type === 'outline' ? StyleSheet.hairlineWidth : 0,
            },
            buttonStyle,
            disabled &&
              type === 'solid' && {
                backgroundColor: theme?.colors?.disabled,
              },
            disabled &&
              type === 'outline' && {
                borderColor: color(theme?.colors?.disabled)
                  .darken(0.3)
                  .string(),
              },
            disabled && disabledStyle,
          ])}
        >
          {/* Activity Indicator on loading */}
          {loading && (
            <ActivityIndicator
              style={StyleSheet.flatten([styles.loading, loadingStyle])}
              color={loadingProps.color}
              size={loadingProps.size}
              {...loadingProps}
            />
          )}
          {/* Button Icon, hide Icon while loading */}
          {!loading &&
            icon &&
            renderNode(Icon, icon, {
              containerStyle: StyleSheet.flatten([
                styles.iconContainer,
                iconContainerStyle,
              ]),
            })}
          {/* Title for Button, hide while loading */}
          {!loading &&
            React.Children.toArray(children).map((child, index) => (
              <React.Fragment key={index}>
                {typeof child === 'string'
                  ? renderNode(Text, child, {
                      style: {
                        ...titleStyle,
                      },
                      ...titleProps,
                    })
                  : child}
              </React.Fragment>
            ))}
        </ViewComponent>
      </TouchableComponentInternal>
    </View>
  );
};

const styles = StyleSheet.create({
  button: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    padding: defaultTheme.spacing.md,
    paddingHorizontal: defaultTheme.spacing.lg,
  },
  container: {
    overflow: 'hidden',
  },
  title: {
    fontSize: 16,
    textAlign: 'center',
    paddingVertical: 1,
    ...Platform.select({
      android: {
        fontFamily: 'sans-serif-medium',
      },
      default: {
        fontSize: 18,
      },
    }),
  },
  iconContainer: {
    marginHorizontal: 5,
  },
  raised: {
    backgroundColor: '#fff',
    overflow: 'visible',
    ...Platform.select({
      android: {
        elevation: 4,
      },
      default: {
        shadowColor: 'rgba(0,0,0, .4)',
        shadowOffset: { height: 1, width: 1 },
        shadowOpacity: 1,
        shadowRadius: 1,
      },
    }),
  },
  loading: {
    marginVertical: 2,
  },
});

Button.displayName = 'Button';
