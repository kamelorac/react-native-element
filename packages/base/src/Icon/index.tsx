import { Icon, IconProps, IconNode, IconObject, IconType } from './Icon';

export { Icon };
export type { IconProps, IconNode, IconObject, IconType };
