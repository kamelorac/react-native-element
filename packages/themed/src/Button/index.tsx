import { withTheme } from '../config';
import { Button, ButtonProps } from '@rneui/base/dist/Button/Button';

export { Button };
export type { ButtonProps };
export default withTheme<ButtonProps>(Button, 'Button');
