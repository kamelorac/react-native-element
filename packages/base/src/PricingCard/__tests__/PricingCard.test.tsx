import React from 'react';
import { PricingCard } from '../index';
import { Icon } from '../../Icon';
import { Button } from '../../Button';
import { renderWithWrapper } from '../../../.ci/testHelper';

describe('PricingCard component', () => {
  it('should match snapshot', () => {
    const component = renderWithWrapper(
      <PricingCard
        info={['1 User', 'Basic Support', 'All Core Features']}
        button={{ title: 'GET STARTED', icon: 'flight-takeoff' }}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render titleStyle', () => {
    const { queryByText } = renderWithWrapper(
      <PricingCard
        info={['1 User', 'Basic Support', 'All Core Features']}
        button={{
          title: 'GET STARTED',
          icon: 'flight-takeoff',
          titleStyle: { fontSize: 30 },
        }}
      />
    );
    expect(queryByText('GET STARTED').props.style).toMatchObject({
      fontSize: 30,
    });
  });

  it('should render with props', () => {
    const component = renderWithWrapper(
      <PricingCard
        title="Free"
        price="$0"
        info={['1 User', 'Basic Support', 'All Core Features']}
        button={{ title: 'GET STARTED', icon: 'flight-takeoff' }}
        titleStyle={{ fontFamily: 'arial' }}
        infoStyle={{ fontFamily: 'arial' }}
        pricingStyle={{ fontFamily: 'arial' }}
        containerStyle={{ backgroundColor: 'peru' }}
        wrapperStyle={{ backgroundColor: 'peru' }}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('button with custom component', () => {
    const { wrapper } = renderWithWrapper(
      <PricingCard
        button={<Button title="GET STARTED" testID="myButton" disabled />}
      />,
      'myButton'
    );
    expect(wrapper).not.toBeNull();
  });

  it('button with extended props', () => {
    const { wrapper } = renderWithWrapper(
      <PricingCard
        button={{
          title: 'GET STARTED',
          icon: { name: 'flight-takeoff', size: 30, color: 'red' },
          disabled: true,
        }}
      />
    );
    expect(wrapper.findByType(Icon).props).toMatchObject({
      name: 'flight-takeoff',
      size: 30,
      color: 'red',
    });
    expect(wrapper.findByType(Button).props).toMatchObject({
      title: 'GET STARTED',
      disabled: true,
    });
  });
});
