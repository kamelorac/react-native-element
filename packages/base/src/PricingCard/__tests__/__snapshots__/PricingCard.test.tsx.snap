// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PricingCard component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      {
        "backgroundColor": "#ffffff",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 15,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 0.5,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "transparent",
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#2089dc",
            "fontSize": 38.57142857142857,
            "fontWeight": "800",
            "textAlign": "center",
          }
        }
        testID="pricingCardTitle"
      />
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#242424",
            "fontSize": 51.42857142857143,
            "fontWeight": "700",
            "marginBottom": 10,
            "marginTop": 10,
            "textAlign": "center",
          }
        }
      />
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#86939e",
            "fontWeight": "600",
            "marginBottom": 5,
            "marginTop": 5,
            "textAlign": "center",
          }
        }
      >
        1 User
      </Text>
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#86939e",
            "fontWeight": "600",
            "marginBottom": 5,
            "marginTop": 5,
            "textAlign": "center",
          }
        }
      >
        Basic Support
      </Text>
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#86939e",
            "fontWeight": "600",
            "marginBottom": 5,
            "marginTop": 5,
            "textAlign": "center",
          }
        }
      >
        All Core Features
      </Text>
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 2,
            },
            undefined,
            false,
          ]
        }
        testID="RNE_BUTTON_WRAPPER"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="RNE__PricingButton"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#2089dc",
                "borderColor": "#2089dc",
                "borderRadius": 2,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 10,
                "marginTop": 15,
                "padding": 8,
                "paddingHorizontal": 10,
              }
            }
          >
            <View
              style={
                {
                  "overflow": "hidden",
                }
              }
              testID="RNE__ICON__CONTAINER"
            >
              <View
                accessibilityRole="button"
                android_ripple={
                  {
                    "borderless": false,
                    "color": "rgba(0, 0, 0, 0.3)",
                    "radius": -5,
                  }
                }
                disabled={false}
                testID="RNE__ICON__CONTAINER_ACTION"
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                  testID="RNE__ICON"
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      [
                        {
                          "color": "white",
                          "fontSize": 15,
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                        {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                    testID="RNE__ICON__Component"
                  >
                    î¤…
                  </Text>
                </View>
              </View>
            </View>
            <Text
              style={
                {
                  "color": "white",
                  "fontSize": 18,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              GET STARTED
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`PricingCard component should render with props 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      {
        "backgroundColor": "peru",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 15,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 0.5,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "peru",
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#2089dc",
            "fontFamily": "arial",
            "fontSize": 38.57142857142857,
            "fontWeight": "800",
            "textAlign": "center",
          }
        }
        testID="pricingCardTitle"
      >
        Free
      </Text>
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#242424",
            "fontFamily": "arial",
            "fontSize": 51.42857142857143,
            "fontWeight": "700",
            "marginBottom": 10,
            "marginTop": 10,
            "textAlign": "center",
          }
        }
      >
        $0
      </Text>
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#86939e",
            "fontFamily": "arial",
            "fontWeight": "600",
            "marginBottom": 5,
            "marginTop": 5,
            "textAlign": "center",
          }
        }
      >
        1 User
      </Text>
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#86939e",
            "fontFamily": "arial",
            "fontWeight": "600",
            "marginBottom": 5,
            "marginTop": 5,
            "textAlign": "center",
          }
        }
      >
        Basic Support
      </Text>
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#86939e",
            "fontFamily": "arial",
            "fontWeight": "600",
            "marginBottom": 5,
            "marginTop": 5,
            "textAlign": "center",
          }
        }
      >
        All Core Features
      </Text>
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 2,
            },
            undefined,
            false,
          ]
        }
        testID="RNE_BUTTON_WRAPPER"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="RNE__PricingButton"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#2089dc",
                "borderColor": "#2089dc",
                "borderRadius": 2,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 10,
                "marginTop": 15,
                "padding": 8,
                "paddingHorizontal": 10,
              }
            }
          >
            <View
              style={
                {
                  "overflow": "hidden",
                }
              }
              testID="RNE__ICON__CONTAINER"
            >
              <View
                accessibilityRole="button"
                android_ripple={
                  {
                    "borderless": false,
                    "color": "rgba(0, 0, 0, 0.3)",
                    "radius": -5,
                  }
                }
                disabled={false}
                testID="RNE__ICON__CONTAINER_ACTION"
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                  testID="RNE__ICON"
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      [
                        {
                          "color": "white",
                          "fontSize": 15,
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                        {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                    testID="RNE__ICON__Component"
                  >
                    î¤…
                  </Text>
                </View>
              </View>
            </View>
            <Text
              style={
                {
                  "color": "white",
                  "fontSize": 18,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              GET STARTED
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
